graylog2-web-interface/src/components/graylog/Button.jsx

```diff
@@ -1,13 +1,78 @@
-import React, { forwardRef } from 'react';
+import React from 'react';
 import styled, { css } from 'styled-components';
 // eslint-disable-next-line no-restricted-imports
 import { Button as BootstrapButton } from 'react-bootstrap';
 
-const StyledButton = styled(BootstrapButton)(({ theme }) => css`
-  ${theme.components.button}
-`);
+const buttonStyles = (bsStyle = 'default') => css(({ theme }) => {
```

Comment:
I do think it's still unnecessary to compute styles over and over on every initial render of a button component, especially considering that the possible values of `bsStyle` is a rather small set. Using `memo` improves re-rendering times, computing button styles once and stuffing computed styles into the theme would also improve the initial rendering.